File: backend/Models/Answer.cs
using System.ComponentModel.DataAnnotations;

namespace prid.Models;

public class Answer {
    [Key]
    public int Id { get; set; }
    public string Sql { get; set; } = "";
    public DateTimeOffset Timestamp { get; set; }
    public bool IsCorrect { get; set; } = false;
    public Question Question { get; set; } = null!;
    public int QuestionId { get; set; }
    public Attempt Attempt { get; set; } = null!;
    public int AttemptId { get; set; }
}

public class AnswerDto {
    public int Id { get; set; }
    public string Sql { get; set; } = "";
    public DateTimeOffset Timestamp { get; set; }
    public bool IsCorrect { get; set; } = false;
    public int QuestionId { get; set; }
    public int AttemptId { get; set; }
    
}

public class EvalDto {
    public string Sql { get; set; } = "";
    public int QuestionId { get; set; }
    public bool isDisplay { get; set; } = false;

}

File: frontend/src/app/models/answer.ts
import { Type } from 'class-transformer';
import { format } from 'date-fns';

export class Answer {
    id?: number;
    sql?: string;
    @Type(() => Date)
    timestamp?: Date;
    isCorrect?: boolean;
    questionId?: number;
    attemptId?: number;

    get getTimeStamp(): string {
        return this.timestamp !== null && this.timestamp? format(this.timestamp, 'dd/MM/yyyy HH:mm:ss') : 'N/A';
    }
}